modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientAuthenticationTest.java

```diff
@@ -0,0 +1,174 @@
+import static org.apache.ignite.internal.testframework.matchers.CompletableFutureMatcher.willCompleteSuccessfully;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+import com.typesafe.config.ConfigFactory;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.client.BasicAuthenticator;
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.internal.app.IgniteImpl;
+import org.apache.ignite.internal.configuration.ConfigurationRegistry;
+import org.apache.ignite.internal.security.authentication.basic.BasicAuthenticationProviderChange;
+import org.apache.ignite.internal.security.configuration.SecurityConfiguration;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.security.exception.InvalidCredentialsException;
+import org.apache.ignite.sql.Session;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Thin client authentication test.
+ */
+public class ItThinClientAuthenticationTest extends ItAbstractThinClientTest {
+    private static final String USERNAME = "admin";
+
+    private static final String PASSWORD = "password";
+
+    /** Client. */
+    private IgniteClient clientWithAuth;
+
+    private SecurityConfiguration securityConfiguration;
+
+    private final BasicAuthenticator basicAuthenticator = BasicAuthenticator.builder()
+            .username(USERNAME)
+            .password(PASSWORD)
+            .build();
+
+    @BeforeEach
+    void setUp() {
+        securityConfiguration = clusterConfigurationRegistry().getConfiguration(SecurityConfiguration.KEY);
+
+        CompletableFuture<Void> enableAuthentication = securityConfiguration.change(change -> {
+            change.changeEnabled(true);
+            change.changeAuthentication()
+                    .changeProviders()
+                    .update("default", provider -> {
+                        provider.convert(BasicAuthenticationProviderChange.class)
+                                .changeUsers()
+                                .createOrUpdate(USERNAME, user -> user.changePassword(PASSWORD));
+                    });
+        });
+
+        assertThat(enableAuthentication, willCompleteSuccessfully());
+
+        clientWithAuth = IgniteClient.builder()
+                .authenticator(basicAuthenticator)
+                .reconnectThrottlingRetries(0)
+                .addresses(getClientAddresses().toArray(new String[0]))
+                .build();
+
+        await().untilAsserted(() -> checkConnection(clientWithAuth));
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        IgniteUtils.closeAll(clientWithAuth);
+    }
+
+    @Test
+    void connectionIsClosedIfAuthenticationEnabled() {
+        await().until(() -> checkConnection(client()), willThrowWithCauseOrSuppressed(InvalidCredentialsException.class));
```

Comment:
Can you add a case when you change the configuration (add one more provider or user) and the connection is still be opened.